`HifitimeError` type appears not to implement `core::error::Error` in `no_std`

When compiled in `no_std` mode (i.e. without the default `std` feature, it appears that the `hifitime::error::HifitimeError` type doesn't implement `core::error::Error`, which causes compilation errors:
```zsh
% cargo check --no-default-features
error[E0599]: the method `as_dyn_error` exists for reference `&HifitimeError`, but its trait bounds were not satisfied
  --> src/lib.rs:5:13
   |
 5 |     #[error(transparent)]
   |             ^^^^^^^^^^^ method cannot be called on `&HifitimeError` due to unsatisfied trait bounds
   |
  ::: /home/amodam/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/hifitime-4.2.3/src/errors.rs:25:1
   |
25 | pub enum HifitimeError {
   | ---------------------- doesn't satisfy `HifitimeError: AsDynError<'_>` or `HifitimeError: core::error::Error`
   |
   = note: the following trait bounds were not satisfied:
           `HifitimeError: core::error::Error`
           which is required by `HifitimeError: AsDynError<'_>`
           `&HifitimeError: core::error::Error`
           which is required by `&HifitimeError: AsDynError<'_>`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `repro` (lib) due to 1 previous error
```

### Repro Instructions

You can repro this with a minimal example:

```rust
/// file: Cargo.toml
[package]
name = "repro"
version = "0.1.0"
edition = "2024"

[features]
default = ["std"]
std = ["hifitime/std", "thiserror/std"]

[dependencies]
hifitime = { version = "4.2.3", default-features = false }
thiserror = { version = "2.0.17", default-features = false }

/// file: src/lib.rs
#![cfg_attr(not(feature = "std"), no_std)]

#[derive(Debug, thiserror::Error)]
pub enum Error {
    #[error(transparent)]
    Hifitime(#[from] hifitime::errors::HifitimeError),
}
```

Compiling with default features works great, compiling without (`cargo c --no-default-features`) fails.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -42,7 +42,7 @@ lexical-core = { version = "1.0.1", default-features = false, features = [
 tabled = { version = "0.20.0", optional = true }
 openssl = { version = "0.10", features = ["vendored"], optional = true }
 web-time = { version = "1.0.0", optional = true }
-snafu = { version = "0.8.2", default-features = false }
+snafu = { version = "0.8.5", features = ["rust_1_81"], default-features = false }
 ureq = { version = "3.0.10", default-features = false, optional = true, features = [
     "rustls",
 ] }

